packages/get-platform/src/getPlatform.ts

```diff
@@ -37,15 +37,28 @@ export type DistroInfo = {
    */
   targetDistro?: 'rhel' | 'debian' | 'musl' | 'arm' | 'nixos' | 'freebsd11' | 'freebsd12' | 'freebsd13'
 }
-export type GetOSResult = {
-  platform: NodeJS.Platform
+type GetOsResultLinux = {
+  platform: 'linux'
   arch: Arch
+  archFromUname: string | undefined
   /**
    * Starting from version 3.0, OpenSSL is basically adopting semver, and will be API and ABI compatible within a major version.
    */
   libssl?: typeof supportedLibSSLVersions[number]
 } & DistroInfo
 
+export type GetOSResult =
+  | {
+      platform: Omit<NodeJS.Platform, 'linux'>
+      arch: Arch
+      targetDistro?: DistroInfo['targetDistro']
+      familyDistro?: never
+      originalDistro?: never
+      archFromUname?: never
+      libssl?: never
+    }
+  | GetOsResultLinux
```

Comment:
This is now a discriminated union on the `platform` property.
This was necessary to move the `alpine:unsupported-arch` into `getPlatformInternal` in a type-safe fashion.